docs/user/guides/hosting-integrations.rst

```diff
@@ -0,0 +1,97 @@
+Hosting Integrations
+====================
+
+Read the Docs performs some actions by default when building documentation for its users to.
+These actions integrate the documentation project with multiple hosting features Read the Docs offers.
+However, when using ``build.commands`` configuration (see :doc:`build-customization`),
+the build process is overriden completely and it's the user responsibility to integrate these features.
+This page explains how to manually integrate these features when Read the Docs cannot do it automatically.
+
+These are all the hosting features that Read the Docs provides by default:
+
+`Flyout menu`_
+   Menu displaying on all documentation pages containing the list of active versions,
+   downloadable formats and other useful links.
+   Read more at :doc:`flyout-menu`.
+
+`Search index`_
+   Smart search index powered by Elasticsearch able to differentiate between regular text,
+   function names in code, and search in all subprojects, among others.
+   Read more at :doc:`server-side-search`.
+
+`Search as you type`_
+   Level up your search by providing immediate feedback while the user is typing.
+   Read more at :doc:`advanced-search.html#enable-search-as-you-type-in-your-documentation`.
+
+`Version warning`_
```

Comment:
Hrm, I thought we had an extension for this?

Reply:
We have a core feature that handles the common use case _and_ a Sphinx extension that's more configurable. However, IIRC it broke some time ago due to CORS changes.